### PR TITLE
E2K: added initial support of MCST Elbrus 2000 CPU architecture

### PR DESCRIPTION
**Pull request by EntityFX**
E2K (Elbrus 2000) - this is VLIW/EPIC architecture, like Intel Itanium (IA-64) architecture.

Ref: https://en.wikipedia.org/wiki/Elbrus_2000
image

### CODE REVIEW

File: includes/e2k/processor-platform.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+ *    HardInfo - Displays System Information
+ *    Copyright (C) 2003-2006 Leandro A. F. Pereira <leandro@hardinfo.org>
```
Comment (lpereira) [inline includes/e2k/processor-platform.h:3]: I didn't write this.  You can put your name here instead.

File: modules/devices/e2k/processor.c [hunk 2]
```diff
@@ -0,0 +1,387 @@
+/*
+ *    HardInfo - Displays System Information
+ *    Copyright (C) 2003-2006 Leandro A. F. Pereira <leandro@hardinfo.org>
```
Comment (lpereira) [inline modules/devices/e2k/processor.c:3]: Same thing.

File: modules/devices/e2k/processor.c [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+#define PROC_SCAN_READ_BUFFER_SIZE 1024
+GSList *processor_scan(void)
+{
+    GSList *procs = NULL, *l = NULL;
+    Processor *processor = NULL;
+    FILE *cpuinfo;
+    gchar *buffer;
+
+    buffer = g_malloc(PROC_SCAN_READ_BUFFER_SIZE);
```
Comment (lpereira) [inline modules/devices/e2k/processor.c:250]: Buffer is never freed; might be fine to just use a `char buffer[1024]`.

File: modules/devices/e2k/processor.c [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+    cpuinfo = fopen(PROC_CPUINFO, "r");
+    if (!cpuinfo)
+        return NULL;
+
+    while (fgets(buffer, PROC_SCAN_READ_BUFFER_SIZE, cpuinfo)) {
+        int rlen = strlen(buffer);
```
Comment (lpereira) [inline modules/devices/e2k/processor.c:256]: `fgets()` already returns the number of bytes read in that line, so you don't need to call `strlen()` here.

File: includes/e2k/processor-platform.h (hunk 1 above)
Comment (EntityFX) [reply includes/e2k/processor-platform.h]: Fixed

File: modules/devices/e2k/processor.c (hunk 2 above)
Comment (EntityFX) [reply modules/devices/e2k/processor.c]: Fixed

File: modules/devices/e2k/processor.c (hunk 3 above)
Comment (EntityFX) [reply modules/devices/e2k/processor.c]: Defined as `char buffer[1024]`

File: modules/devices/e2k/processor.c (hunk 4 above)
Comment (EntityFX) [reply modules/devices/e2k/processor.c]: Defined as char buffer[1024]

**Comment by lpereira** [comment]
Merged, thanks!

**Comment by EntityFX** [comment]
https://translate.google.com/translate?hl=ru&sl=ru&tl=en&u=http%3A%2F%2Fftp.altlinux.org%2Fpub%2Fpeople%2Fmike%2Felbrus%2Fdocs%2Felbrus_prog%2Fhtml%2F

**Pull request merged**